reconcile/aws_cloudwatch_log_retention/integration.py

```diff
@@ -112,3 +124,18 @@ def run(dry_run: bool, thread_pool_size: int, defer: Optional[Callable] = None)
                                         cloudwatch_cleanup_entry.log_retention_day_length
                                     ),
                                 )
+        try:
+            awsapi = create_awsapi_client(aws_acct)
+            log_group_list = create_log_group_list(awsapi, aws_acct)
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "AccessDeniedException":
+                logging.info(f" Access denied for {aws_act_name}. Skipping...")
+                continue
+        else:
+            for log_group in log_group_list:
+                group_name = log_group["logGroupName"]
+                retention_days = log_group.get("retentionInDays")
+                if retention_days is None:
+                    logging.info(f" Setting {group_name} retention days to 90")
+                    if not dry_run:
+                        awsapi.set_cloudwatch_log_retention(aws_acct, group_name, 90)
```

Comment:
There are duplicate logic in this block with `cleanup` block above, it seems what you want is

* for each aws account
  * get all log groups `create_log_group_list` (should rename to `get_log_group_list` as we are not creating new resources)
  * for each log group
    * find matching `cloudwatch_cleanup_entry` 
    * if find
      * `create_cloudwatch_tag` and `set_cloudwatch_log_retention` accordingly
    * else
      * `set_cloudwatch_log_retention` with default value

If you follow above logic and extract methods for each section, it should be easier to understand.

A few tips:
* `log_regex` in `AWSCloudwatchLogRetention` can be `typing.Pattern` directly (`log_regex=re.compile(aws_acct_field["regex"]),`) to avoid recompile multiple times
* to find a matching `cloudwatch_cleanup_entry`, can use this (`cloudwatch_cleanup_entry` will be `None` if not found)
  ```python
                       cloudwatch_cleanup_entry = next(
                            (
                                c
                                for c in cloudwatch_cleanup_list
                                if c.log_regex.match(group_name)
                            ),
                            None,
                        )
  ```